File: Sorting/InsertionSort/java/InsertionSort.java
import java.util.Scanner;

public class InsertionSort {

    public static void input(int arr[], int size) {
        System.out.println("Enter the array: ");
        Scanner in = new Scanner(System.in);
        for(int i=0; i<size; i++) {
            arr[i] = in.nextInt();
        }
        in.close();
    }

    public static void insertionSort(int arr[], int size) {
        int temp, empty_index;
        for(int i=1; i<size; i++) {
            temp = arr[i];
            empty_index = i;
            while(empty_index > 0 && arr[empty_index-1] > temp) {
                arr[empty_index] = arr[empty_index-1];
                empty_index --;
            }
            arr[empty_index] = temp;
        }
    }

    public static void printArray(int[] arr, int size) {
        System.out.print("Array is: ");
        for(int i=0; i<size; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }
}

File: Algorithms/Sorting/Insertion Sort/python/InsertionSort.py
def insertionSort(arr):
    for i in range(1, len(arr)):
        temp = arr[i]
        empty_index = i
        while empty_index > 0  and arr[empty_index-1] > temp:
            arr[empty_index] = arr[empty_index-1]
            empty_index -= 1
        arr[empty_index] = temp


if __name__ == '__main__':
    arr = list(map(int, input("Enter the array: ").split()))
    print("Array is: " + " ".join([str(i) for i in arr]))
    print("Sorting...")
    insertionSort(arr)
    print("Array is: " + " ".join([str(i) for i in arr]))